tools/gfx/render.h

```diff
@@ -1,28 +1,36 @@
 // render.h
 #pragma once
 
-//#include "shader-input-layout.h"
-
 #include <float.h>
-
-#include "../../slang-com-helper.h"
-
-#include "../../source/core/slang-smart-pointer.h"
-#include "../../source/core/slang-list.h"
-#include "../../source/core/slang-dictionary.h"
-#include "../../source/core/slang-process-util.h"
-
-#include "../../slang.h"
+#include <assert.h>
+
+#include "slang.h"
+#include "slang-com-ptr.h"
+#include "slang-com-helper.h"
+
+
+#if defined(SLANG_GFX_DYNAMIC)
+#    if defined(_MSC_VER)
+#        ifdef SLANG_GFX_DYNAMIC_EXPORT
+#            define SLANG_GFX_API SLANG_DLL_EXPORT
+#        else
+#            define SLANG_GFX_API __declspec(dllimport)
+#        endif
+#    else
+// TODO: need to consider compiler capabilities
+//#     ifdef SLANG_DYNAMIC_EXPORT
+#        define SLANG_GFX_API SLANG_DLL_EXPORT
+//#     endif
+#    endif
+#endif
+
+#ifndef SLANG_GFX_API
+#    define SLANG_GFX_API
+#endif
 
 namespace gfx {
 
-using Slang::RefObject;
-using Slang::RefPtr;
-using Slang::Dictionary;
-using Slang::List;
-
-using Slang::getHashCode;
-using Slang::combineHash;
+using Slang::ComPtr;
```

Comment:
We may eventually want to make the dependency on `Slang::ComPtr` optional, but we can handle that down the line.